app/src/main/java/unipd/se18/ocrcamera/ResultActivity.java

```diff
@@ -0,0 +1,361 @@
+     * @return String with the list of ingredients found in the text
+     * Put in database csv: https://www.youtube.com/watch?v=i-TqNzUryn8
+     */
+    public String inciDetector(String text){
+        String inci = "";
+        try {
+            InputStream is = getResources().openRawResource(R.raw.database);    //InputStream from database.csv
+            BufferedReader reader = new BufferedReader(new InputStreamReader(is, Charset.forName("UTF-8")));    //database reader
+            String ingredient;  //String of inci ingredient
+            while ((ingredient=reader.readLine())!= null){      //defines ingredient as each row of the database
+                if (text.toUpperCase().contains(ingredient)) {      //check if no text found are inci ingredients
+                    inci = inci + "<b>" + ingredient + "</b>; ";    //Create the string with the ingredients in bold
+                }
+            }
+        }catch (IOException e){
+            e.printStackTrace();
+        }
+
+        return inci;
+    }
+
+    /**
+     * @author Giovanni Fasan(g1), Giovanni Piva(g1)
+     * @param text String in which you have to find the ingredients
+     * @return String with ingredients list of the label
+     * Put in database csv: https://www.youtube.com/watch?v=i-TqNzUryn8
+     * problem contains: from the string "black" find the ingredient "LAC"
+     * the problem is solved by a division of the string into characters to which I am still workings
+     */
+    public String inciDetectorEtichetta(String text){
+        String inci = "";
+        List listInci = new ArrayList();        //list where save the ingredients found
+
+        try {
+            InputStream is = getResources().openRawResource(R.raw.database);    //InputStream from database.csv
+            is.mark(0);     //mark the position to restart the buffer from line 0 of database
+            BufferedReader reader = new BufferedReader(new InputStreamReader(is, Charset.forName("UTF-8")));    //database reader
+            String ingredient;      //String of inci ingredient
+            String noSpace = text.replaceAll("\n", " ");    //remove \n (start a new line) and replace with " ".
+            String[] tokens = noSpace.split(",");   //split the text by "," for find the inci ingredients
+            String box="";
+
+            for (int i=0; i<tokens.length; i++) {
+                while ((ingredient=reader.readLine())!=null){       //defines ingredient as each row of the database
+                    if (tokens[i].toUpperCase().contains(ingredient)) {         //Use .toUpperCase because the database's words are uppercase. .contains find if tokens[i] is an ingredient
+                        box=ingredient;     //save the ingredient in a box. after while i use the last ingredient that I found
+                    }
+                }
+                is.reset();     //reset InputStream to the mark 0
+                if (!listInci.contains(box)) {
```

Comment:
If you don't want duplicates, you may consider using a set data structure that by default checks for duplicates when you try to add an item to the set.